cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,266 @@
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    monitoring = Dropdown('Monitoring')
+    download = Button(title='Download summary in PDF format')
+
+
+class DeploymentRoleComparisonToolbar(View):
+    """The toolbar on Comparison Page of roles"""
+    all_attributes = Button(title="All attributes")
+    different_values_attributes = Button(title="Attributes with different values")
+    same_values_attributes = Button(title="Attributes with same values")
+
+    details_mode = Button(title="Details Mode")
+    exists_mode = Button(title="Exists Mode")
+    download = Dropdown('Download')
+
+
+class DeploymentRoleDetailsAccordion(View):
+    """The accordion on the Deployment Role details page"""
+    @View.nested
+    class properties(Accordion):        # noqa
+        nav = BootstrapNav('//div[@id="ems_prop"]//ul')
+
+    @View.nested
+    class relationships(Accordion):     # noqa
+        nav = BootstrapNav('//div[@id="ems_rel"]//ul')
+
+
+class DRQuadIconEntity(BaseQuadIconEntity):
+    @property
+    def ems_cluster(self):
+        return self.browser.get_attribute("alt", self.QUADRANT.format(pos="a"))
+
 
-cfg_btn = partial(tb.select, 'Configuration')
-pol_btn = partial(tb.select, 'Policy')
-match_page = partial(match_location, controller='ems_cluster')
+class DRTileIconEntity(BaseTileIconEntity):
+    quad_icon = ParametrizedView.nested(DRQuadIconEntity)
 
 
-class DeploymentRoles(Pretty, Navigatable):
+class DRListEntity(BaseListEntity):
+    pass
+
+
+class NonJSDREntity(NonJSBaseEntity):
+    quad_entity = DRQuadIconEntity
+    list_entity = DRListEntity
+    tile_entity = DRTileIconEntity
+
+
+def DeploymentRolesEntities():  # noqa
+    """Temporary wrapper for Deployment Role Entity during transition to JS based Entity """
+    return VersionPick({
+        Version.lowest(): NonJSDREntity,
+        '5.9': JSBaseEntity,
+    })
+
+
+class DeploymentRoleEntitiesView(BaseEntitiesView):
+    """The entities on the main list Deployment Role page"""
+    @property
+    def entity_class(self):
+        return DeploymentRolesEntities().pick(self.browser.product_version)
+
+
+class DeploymentRoleDetailsEntities(View):
+    """The entities on the Deployment Role details page"""
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    relationships = SummaryTable(title='Relationships')
+    total_for_node = SummaryTable(title='Totals for Nodes')
+    total_for_vm = SummaryTable(title='Totals for VMs')
+    smart_management = SummaryTable(title='Smart Management')
+
+
+class DeploymentRoleComparisonEntities(View):
+    """The entities on compare Deployment role page"""
+    breadcrumb = BreadCrumb()
+    title = Text('//div[@id="main-content"]//h1')
+    table = Table('//*[@id="compare-grid"]/table')
+
+
+class DeploymentRoleView(BaseLoggedInPage):
+    """A base view for all the Deployment Role pages"""
+    title = Text('.//div[@id="center_div" or @id="main-content"]//h1')
```

Comment:
as far as I noticed you have title in each entities view. So, you don't need title here.

Reply:
I kept it but removed the excessive use of the title.